services/catalog/src/simcore_service_catalog/api/routes/services_resources.py

```diff
@@ -0,0 +1,125 @@
+    # get the service resources
+    service_resources = default_service_resources.copy(deep=True)
+    for entry in resource_entries:
+        if not isinstance(entry.value, dict):
+            logger.warning(
+                "resource %s for %s got invalid type",
+                f"{entry.dict()!r}",
+                f"{service_key}:{service_version}",
+            )
+            continue
+        if nano_cpu_limit := entry.value.get("Limits", {}).get("NanoCPUs"):
+            service_resources["CPU"].limit = nano_cpu_limit / 1.0e09
+        if nano_cpu_reservation := entry.value.get("Reservations", {}).get("NanoCPUs"):
+            service_resources["CPU"].reservation = nano_cpu_reservation / 1.0e09
+        if ram_limit := entry.value.get("Limits", {}).get("MemoryBytes"):
+            service_resources["RAM"].limit = ram_limit
+        if ram_reservation := entry.value.get("Reservations", {}).get("MemoryBytes"):
+            service_resources["RAM"].reservation = ram_reservation
+
+        if generic_resources := entry.value.get("Reservations", {}).get(
+            "GenericResources", []
+        ):
+            for res in generic_resources:
+                if not isinstance(res, dict):
+                    continue
+
+                if named_resource_spec := res.get("NamedResourceSpec"):
+                    service_resources.setdefault(
+                        named_resource_spec["Kind"],
+                        ResourceValue(
+                            limit=0, reservation=named_resource_spec["Value"]
+                        ),
+                    ).reservation = named_resource_spec["Value"]
+                if discrete_resource_spec := res.get("DiscreteResourceSpec"):
+                    service_resources.setdefault(
+                        discrete_resource_spec["Kind"],
+                        ResourceValue(
+                            limit=0, reservation=discrete_resource_spec["Value"]
+                        ),
+                    ).reservation = discrete_resource_spec["Value"]
+
+    return service_resources
+
+
+@router.get(
+    "/{service_key:path}/{service_version}/resources",
+    response_model=dict[ResourceName, ResourceValue],
+    **RESPONSE_MODEL_POLICY,
+)
+@cached(
+    ttl=DIRECTOR_CACHING_TTL,
+    key_builder=lambda f, *args, **kwargs: f"{f.__name__}_{kwargs['service_key']}_{kwargs['service_version']}",
+)
+async def get_service_resources(
+    service_key: ServiceKey,
+    service_version: ServiceVersion,
+    director_client: DirectorApi = Depends(get_director_api),
+    default_service_resources: ServiceResources = Depends(
+        get_default_service_resources
+    ),
+):
+    # cause these entries are not in ServiceDockerData
+    if is_function_service(service_key):
+        # NOTE: this is due to the fact that FastAPI does not appear to like pydantic models with just a root
+        return default_service_resources.dict()["__root__"]
```

Comment:
As I commented in https://github.com/ITISFoundation/osparc-simcore/pull/2988, 
IMO we should go for ``ResourcesDict = dict[ResourceName, ResourceValue]`` instead of  ``DictModel[ResourceName, ResourceValue]`` and then
use ``parse_obj_as(ResourcesDict, obj)`` for FastAPI responses.
Can you give me an example where ``DictModel`` is more advantageous wrt ``dict``+ ``parse_obj_as``??

Reply:
I think I do agree. but I would move that to a separate refactoring effort...